guardian/frontend
# Okta: Header sign in status

Co-authored-by: Jamie <53781962+JamieB-gu@users.noreply.github.com>

This PR continues the Okta migration. In order to keep the PR as small as possible, this change only affects the header's signed in status:

<img width="515" alt="image" src="https://github.com/guardian/frontend/assets/705427/e8f9e8cc-5050-48b2-993d-6619d2e62ffc">

During the migration to Okta, we need to support two methods of determining whether a user is signed in: the original method using cookies, and the new method using Okta.

We first determine if the user is enrolled in the Okta experiment. If they are not, we fall back to the existing cookie implementation.

If they are, we:
- Lazy load the new `okta` module introduced in this PR. This module loads the new `@guardian/identity-auth` dependency

## What has changed?

- Add @guardian/identity-auth as a dependency
- Add an okta module, which is lazy loaded if the user is enrolled in the Okta experiment
- Add a isUserLoggedInRefactor method to the api module which:
  - checks if the user is enrolled in the Okta experiment. If you are in the experiment, use the Okta methods to determine sign in status. Otherwise, fall back to the existing cookie-based method
- Introduce an `AuthStatus` type to model whether a user is signed in/out with cookies, or signed in/out with Okta 

## How to test

- On CODE. You can opt in to the Okta experiment by visiting https://m.code.dev-theguardian.com/opt/in/okta
- See document [Testing Okta in Development](https://docs.google.com/document/d/1IxHLEK7tx1EOkibToiHPMqOWZr-ZCyr_G__k5wiORgM/edit)

Resolves https://github.com/guardian/frontend/issues/26314
